drivers/adc/ad7768-1/ad77681.c

```diff
@@ -288,6 +317,273 @@ int32_t ad77681_spi_read_adc_data(struct ad77681_dev *dev,
+	/* 16bit format */
+	else
+		measured_data->raw_data[measured_data->count] = (buf[0] << 8);
+
+	/* If status bit or CRC is enabled, handle it */
+	if (dev->status_bit || (dev->crc_sel != AD77681_NO_CRC))
+		ret |= ad77681_CRC_status_handling(dev, buf);
+
+	return ret;
+}
+
+/**
+ * CRC and status bit handling after each readout form the ADC
+ * @param dev - The device structure.
+ * @param *data_buffer - 16-bit buffer readed from the ADC containing the CRC,
+ * data and the stattus bit.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_CRC_status_handling(struct ad77681_dev *dev,
+				    uint16_t *data_buffer)
+{
+	int32_t ret = 0;
+	uint8_t status_byte = 0, checksum = 0, checksum_byte = 0, checksum_buf[5],
+		checksum_length = 0, i;
+	char print_buf[50];
+
+	/* Status bit handling */
+	if (dev->status_bit) {
+		/* 24bit ADC data + 8bit of status = 2 16bit frames */
+		if (dev->conv_len == AD77681_CONV_24BIT)
+			status_byte = data_buffer[1] & 0xFF;
+		/* 16bit ADC data + 8bit of status = 2 16bit frames */
+		else
+			status_byte = data_buffer[1] >> 8;
+	}
+
+	/* Checksum bit handling */
+	if (dev->crc_sel != AD77681_NO_CRC) {
+		if ((dev->status_bit == true) & (dev->conv_len == AD77681_CONV_24BIT)) {
+			/* 24bit ADC data + 8bit of status + 8bit of CRC = 3 16bit frames */
+			checksum_byte = data_buffer[2] >> 8;
+			checksum_length = 4;
+		} else if ((dev->status_bit == true) & (dev->conv_len == AD77681_CONV_16BIT)) {
+			/* 16bit ADC data + 8bit of status + 8bit of CRC = 2 16bit frames */
+			checksum_byte = data_buffer[1] & 0xFF;
+			checksum_length = 3;
+		} else if ((dev->status_bit == false) & (dev->conv_len == AD77681_CONV_24BIT)) {
+			/* 24bit ADC data + 8bit of CRC = 2 16bit frames */
+			checksum_byte = data_buffer[1] & 0xFF;
+			checksum_length = 3;
+		} else if ((dev->status_bit == false) & (dev->conv_len == AD77681_CONV_16BIT)) {
+			/* 16bit ADC data + 8bit of CRC = 2 16bit frames */
+			checksum_byte = data_buffer[1] >> 8;
+			checksum_length = 2;
+		}
+
+		for (i = 0; i < checksum_length; i++) {
+			if (i % 2)
+				checksum_buf[i] = data_buffer[i / 2] & 0xFF;
+			else
+				checksum_buf[i] = data_buffer[i / 2] >> 8;
+		}
+
+		if (dev->crc_sel == AD77681_CRC)
+			checksum = ad77681_compute_crc8(checksum_buf, checksum_length,
+							INITIAL_CRC_CRC8);
+		else if (dev->crc_sel == AD77681_XOR)
+			checksum = ad77681_compute_xor(checksum_buf, checksum_length, INITIAL_CRC_XOR);
+
+		if (checksum != checksum_byte)
+			ret = FAILURE;
+
+#ifdef CRC_DEBUG
+
+		char ok[3] = { 'O', 'K' }, fault[6] = { 'F', 'A', 'U', 'L', 'T' };
+		sprintf(print_buf, "\n%x\t%x\t%x\tCRC %s", checksum_byte, checksum, status_byte,
+			((ret == FAILURE) ? (fault) : (ok)));
+		printf(print_buf);
+
+#endif /* CRC_DEBUG */
+	}
+
+	return ret;
+}
+
+/**
+ * Conversion from measured data to voltage
+ * @param dev - The device structure.
+ * @param measured_data - structure carying ADC measurement info
```

Comment:
`ad77681_data_to_voltage` function does not have `measured_data` as parameter. This was also detected by the doxygen build (TravisCI). There are similar issues throughout the code (check the Travis build).

Reply:
Thanks. Done fixing but, no build results from Travis.